### ISSUE TITLE
[Consistency issue: no parent for object `Main Group`] while running pod install

### ISSUE THREAD

**Issue by jcayzac** [CocoaPods/CocoaPods#5847]
### Command

```
/Users/dev/.rbenv/versions/2.3.1/bin/pod install --repo-update --verbose
```

### Report

* What did you do?
Updated my rubygems, then ran the command above in my project.

* What did you expect to happen?
Pods getting installed as they did before updating the rubygems.

* What happened instead?
This crash.

### Stack

```
   CocoaPods : 1.0.1
        Ruby : ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin15]
    RubyGems : 2.5.1
        Host : Mac OS X 10.11.5 (15F34)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /Users/dev/.rbenv/versions/2.3.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 433adab3b71c0209e6005db86a18831e8d39d8b3
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-keys        : 1.7.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
source 'https:/redacted/redacted.git'

platform :ios, '8.0'
use_frameworks!

target 'Tests' do
  pod 'OCMock',          '~> 3.2'
  pod 'OHHTTPStubs',     '~> 4.7.0'
  pod 'OHHTTPStubs/HTTPMessage'

  pod 'RCardInfo', :path => './RCardInfo.podspec'
  pod 'CardIO', '5.3.2'
end
# vim:syntax=ruby:et:sts=2:sw=2:ts=2:ff=unix:
```

### Error

```
RuntimeError - [Xcodeproj] Consistency issue: no parent for object `Main Group`: ``
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:18:in `parent'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:118:in `source_tree_real_path'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:100:in `real_path'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:121:in `source_tree_real_path'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:100:in `real_path'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:121:in `source_tree_real_path'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/groupable_helper.rb:187:in `set_path_with_source_tree'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/file_references_factory.rb:102:in `new_file_reference'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/helpers/file_references_factory.rb:31:in `new_reference'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/xcodeproj-1.3.1/lib/xcodeproj/project/object/group.rb:222:in `new_reference'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:83:in `block in set_target_xcconfig'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:101:in `set_target_xcconfig'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:22:in `block (2 levels) in integrate'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:20:in `each'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:20:in `block in integrate'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:19:in `each'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator/xcconfig_integrator.rb:19:in `integrate'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator.rb:53:in `block in integrate!'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/user_interface.rb:63:in `section'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator/target_integrator.rb:52:in `integrate!'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator.rb:129:in `each'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator.rb:129:in `integrate_user_targets'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer/user_project_integrator.rb:63:in `integrate!'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer.rb:767:in `block in integrate_user_project'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/user_interface.rb:63:in `section'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer.rb:764:in `integrate_user_project'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/installer.rb:120:in `install!'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/command/install.rb:37:in `run'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/lib/cocoapods/command.rb:50:in `run'
/Users/dev/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/cocoapods-1.0.1/bin/pod:55:in `<top (required)>'
/Users/dev/.rbenv/versions/2.3.1/bin/pod:23:in `load'
/Users/dev/.rbenv/versions/2.3.1/bin/pod:23:in `<main>'
```

**Comment by jcayzac** [comment]
I can confirm that if I downgrade `xcodeproj` to `1.3.0`, then I can install my pods. The crash is only with `xcodeproj:1.3.1`.

**Comment by jcayzac** [comment]
I also just confirmed that `cocoapods:1.1.0.rc.1` has the same issue when using `xcodeproj:1.3.1`.

**Comment by jmfieldman** [comment]
No idea if this is related, but I noticed your podfile has:

```source 'https:/redacted/redacted.git'```

instead of

```source 'https://redacted/redacted.git'```

**Comment by tommeier** [comment]
Also getting this, and can confirm is also only present on `xcodeproj` `1.3.0` -> `1.3.1`.

**Comment by p8** [comment]
Can also confirm issue is fixed by downgrading xcodeproj gem to 1.3.0

**Comment by zanesc** [comment]
I am on CocoaPods 1.1.0.rc.2 and i have tried downgrading to xcodeproj gem 1.3.0 and i am still getting this error.

**Comment by tommeier** [comment]
@zanesc remove 1.3.1 entirely, I assume it's defaulting to latest on your setup if you're not running with `bundle exec pod install`

**Comment by zanesc** [comment]
@tommeier i can't on the latest Cocoapods:
```

cocoapods-1.1.0.rc.2 depends on xcodeproj (< 2.0, >= 1.3.1)
If you remove this gem, these dependencies will not be met.
```

**Comment by tommeier** [comment]
@zanesc in which case the issue is still with xcodeproj 1.3.1 (just to clarify)

**Comment by chrisballinger** [comment]
FWIW I'm just seeing this issue as well. Thank you for finding a workaround by downgrading xcodeproj!

**Comment by henkvanderspek** [comment]
@zanesc I'm having the exact same problem as you had. How did you fix it? I can't downgrade xcodeproj for cocoapods 1.1.0 rc 1 or 2.

**Comment by chrisballinger** [comment]
I used a Gemfile/bundler to use cocoapods 1.0.1 / xcodeproj 1.3.0 on the project where it didn't work (it contained no Swift code so didn't need 1.1.0 for Xcode 8).

**Comment by henkvanderspek** [comment]
So, my pod install is still failing with above message. A bit more details about my setup maybe:

I'm trying to migrate my iOS app to Swift 3
I've installed Xcode 8.0 by using a direct download (instead of app store)
This way i was hoping to run Xcode 7.x and 8.0 side by side
I'm using CocoaPods 1.1.0 rc 2
I've tried downgrading to Xcodeproj 1.3.0, but rc 1 or 2 won't let me, because of the dependency
I've tried specifying this in a Gemfile, same result, dependency on Xcodeproj 1.3.1
I've cleared the caches, removed pods dir
If I create a new project in Xcode 8.0, CocoaPods works fine that way
If I run pod install for my Xcode 7.x project, it fails
I've tried this before and after opening in Xcode 8.0
I thought it was because of the extra test target, removing that made no difference

Does it matter that `xcode-select -p` points to Xcode 7.x instead of Xcode 8.0?

**Comment by dhoskins** [comment]
After installing the Cocoapods gem, I currently perform:

    gem uninstall xcodeproj --version 1.3.1
    gem install xcodeproj -v 1.3.0

**Comment by etamity** [comment]
@dhoskins I have tried to revert to 1.3.0, but when I pod install, it throw an error looking for 1.3.1

**Comment by chrisballinger** [comment]
@etamity Make a `Gemfile` in your project root:

```ruby
source "https://rubygems.org"

gem "cocoapods", "1.0.1"
gem "xcodeproj", "1.3.0" # 1.3.1 has a bug https://github.com/CocoaPods/CocoaPods/issues/5847
```

Then run these commands:

```
$ gem install bundler # if bundler is not installed already
$ cd /your/project/directory
$ bundle install
$ bundle exec pod repo update
$ bundle exec pod install
```

etc.

**Comment by henkvanderspek** [comment]
Thanks @chrisballinger! Finally i'm moving again on migrating to Swift 3 👍

**Comment by thinkrad** [comment]
Was running into the same issue. The project was built in Xcode 7. First tried going back to pod 1.0.1 and Xcodeproj 1.3.0. That worked for Swift 2.3, but some of my deps required the latest cocoapods for Swift 3. 

Tried re-creating the project in Xcode 8, and the pod crash disappeared. Luckily, this project was relatively simple to re-create. If you're in a pinch though, this should solve the problem until a bugfix is posted for Xcodeproj.

**Comment by tirodkar** [comment]
Folks, is the `Products` group present in your Xcode Projects? For me the issue was that the `Products` group had been deleted previously.

**Comment by tommeier** [comment]
@tirodkar It was deleted in the past, but its definitely there now? membership to all targets _except_ `UnitTests` (which uses host app)

**Comment by maciejtrybilo** [comment]
It seems this can be fixed on the project side. Looking at a freshly created Xcode project, we can see that `productRefGroup` should be set to an existing group (for new projects it's the "Products" group). In our project for some reason it was set to the same identifier as `mainGroup` which doesn't have  a parent.

Just setting `productRefGroup` to an identifier of an existing group fixes the problem. It is a bit cleaner to make it the "Products one" :)

To recap, it's not about having a "Products" group, but about setting `productRefGroup` correctly.

**Comment by chrisballinger** [comment]
Is this something that could be detected programmatically by CocoaPods and
propose fixing for you? e.g.

We've noticed that productRefGroup is missing a parent group. Would you
like CocoaPods to attempt to fix this? Y/N

Or... perhaps just a workaround in the parser because this is a regression
from xcodeproj 1.3.0 to 1.3.1.

**Comment by tirodkar** [comment]
@tommeier, are you able to pick up the productsRefGroup? Because even after I added such a group to my project, I still faced the same issue. I had to essentially regen my entire project to fix this.

**Comment by tommeier** [comment]
@tirodkar @maciejtrybilo perfect :D that was exactly it.


The 'Products' group seemed to have a ref to the main group itself for `productRefGroup`
```
      mainGroup = 0300ACA3170BE1040015E375;
      productRefGroup = 0300ACA3170BE1040015E375;
      projectDirPath = "";
      projectRoot = "";
      targets = (
        ...objects in Products
      );
    };
```

And the actual ref was totally different:

```
  5C7F27D7C6663D8502EB3D7C /* Pods */,
  E2025B7A1CF7DD47007CAF9B /* Products */,
  E23A4B211A9BD6AD00A4422D /* README.md */,
```

So changing the group to (comment was auto added on next load) match the actual ref: 

```
productRefGroup = E2025B7A1CF7DD47007CAF9B /* Products */;
```

And both `xcodeproj` `1.3.1` + `1.3.2` work flawlessly.

Thanks!!

**Comment by tsheaff** [comment]
the latest xcodeproj is now 1.3.3 which resolves this issue.

`gem update xcodeproj` worked for me

**Comment by kennebecstars** [comment]
@tsheaff that 'gem update xcodeproj' ==> 1.3.3 worked!

```
Updating xcodeproj
Fetching: claide-1.0.1.gem (100%)
Successfully installed claide-1.0.1
Fetching: xcodeproj-1.3.3.gem (100%)
Successfully installed xcodeproj-1.3.3
Gems updated: clad xcodeproj
```

**Comment by jcayzac** [comment]
I'm OP and can confirm that the issue is gone with the latest gems, so I'll just go ahead and close the issue since it was not assigned to anybody yet.
